WWTMVC5/WWTWeb/MartianTile.aspx

```diff
@@ -9,76 +9,64 @@
 
-    string query = Request.Params["Q"];
-    string[] values = query.Split(',');   
+
+ string[] values = query.Split(',');   
     int level = Convert.ToInt32(values[0]);
     int tileX = Convert.ToInt32(values[1]);
     int tileY = Convert.ToInt32(values[2]);
     string dataset = values[3];
-    string id = "nothing";
-	
-  	string DSSTileCache = WWTUtil.GetCurrentConfigShare("DSSTileCache", true);
 
 	switch(dataset)
 	{
 		case "mars_base_map":
-			id = "1738422189";
+			if (level < 18)
+    {
+       // Response.ContentType = "image/png";
+       
+        Stream s = PlateFile2.GetFileStream(Path.Combine(ConfigurationManager.AppSettings["WWTTilesDir"], "marsbasemap.plate"), -1, level, tileX, tileY);
+        
+        if (s == null || (int)s.Length == 0)
+        {
+            Response.Clear();
+            Response.ContentType = "text/plain";
+            Response.Write("No image");
+            Response.End();
+            return;
+        }
+        int length = (int)s.Length;
+        byte[] data = new byte[length];
+        s.Read(data, 0, length);
+        Response.OutputStream.Write(data, 0, length);
+        Response.Flush();
+        Response.End();
+       	 return;
+    	}
 			break;
 		case "mars_terrain_color":
-			id = "220581050";
-			break;
-		case "mars_hirise":
-			id = "109459728";
-			break;
-		case "mars_moc":
-			id = "252927426";
-			break;
-		case "mars_historic_green":
-			id = "1194136815";
-			break;
-		case "mars_historic_schiaparelli":
-			id = "1113282550";
-			break;
-		case "mars_historic_lowell":
-			id = "675790761";
-			break;
-		case "mars_historic_antoniadi":
-			id = "1648157275";
-			break;
-		case "mars_historic_mec1":
-			id = "2141096698";
-			break;
-		
-	}
+		{
+			 WebClient client2 = new WebClient();
 
+    			 byte[] data3 = client2.DownloadData(string.Format("https://wwtfiles.blob.core.windows.net/marsmola/marsmolaL{0}X{1}Y{2}.png",level, tileX, tileY));
+	
+    			 Response.BinaryWrite(data3);
+			return;
+		}
+		break;
+   }
 
-    string filename = String.Format( DSSTileCache  + "\\wwtcache\\mars\\{3}\\{0}\\{2}\\{1}_{2}.png", level, tileX, tileY, id);
-    string path = String.Format(DSSTileCache + "\\wwtcache\\mars\\{3}\\{0}\\{2}", level, tileX, tileY, id);
 
 
-    if (!File.Exists(filename))
-    {
-        //try
-        //{
-        //    if (!Directory.Exists(filename))
-        //    {
-        //        Directory.CreateDirectory(path);
-        //    }
-            
-        //    WebClient webclient = new WebClient();
 
-        //    string url = string.Format("http://wwt.nasa.gov/wwt/p/{0}/{1}/{2}/{3}{4}", dataset, level, tileX, tileY, type);
 
-        //    webclient.DownloadFile(url, filename);
-        //}
-        //catch
-       // {
-            Response.StatusCode = 404;
-            return;
-       // }
-    }
+
+    WebClient client = new WebClient();
+
+    byte[] data2 = client.DownloadData("http://141.142.209.200/wwtweb/martiantile.aspx?q="+query);
```

Comment:
Ah OK, looks like we're hardcoding a fallback to the main server? OK as a stopgap but there should at least be a big "XXX FIXME" comment as a reminder ...

Reply:
That's weird - I thought this was the "main" server? Is that not pointing to the VM? Which also would be weird, but I must be missing something here.

Reply:
Mystery solved, sort of — I didn't realize this was more code from the VMs directories. No idea what this magic IP address is/was, but we're not gonna want it anymore ...